### PR TITLE
feat(gateway): gravitee.yml

### PR DESCRIPTION
**Pull request by sebdevaux**
The goal of this pr is to allow customers to override the configuration of each products by providing their own configmap or secret.

### CODE REVIEW

File: apim/3.x/values.yaml [hunk 1]
```diff
@@ -649,6 +650,16 @@ api:
     #   mountPath: /mnt/volume
     #   readOnly: true
     #   existingClaim: volume-claim
+  # If you want to use your own gravitee.yml you have to provide your configmap or secret in extraVolume part.
+  # the name of the volume MUST be "config".
+  # In this case, values configuration related to gravitee.yaml defined in this file will be ignored
```
[the remark this answers is not in the document]
Comment (sebdevaux) [reply apim/3.x/values.yaml:655]: Not at all, some users want to provide their own configuration as a secret. Today they cannot. With this feature, they can.

[the remark this answers is not in the document]
Comment (ThibaudAV) [reply apim/3.x/values.yaml:655]: Yes, I understand. It's just that I've always found it weird to have this version of gravitee.yml here with default valuers that are not the same as the java projects. 
Okok

**Review by ThibaudAV — APPROVED** [review]
LGTM. Except that some of the files are missing line breaks at the end 🙈 
image

**Review by leleueri — COMMENTED** [review]
Should we create a volume mount section in addition of the volume one?
AFAIK, to mount a secret as a file, we have to define the secret as a volume but also moun this volume. :thinking: Did I miss something ?

File: apim/3.x/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebdevaux) [reply apim/3.x/values.yaml:655]: Unfortunately, we have no other option...

**Comment by sebdevaux** [comment]
The volumeMount is already mounted, that's why the volume Must be named "config"

**Pull request merged**